src/microbe_stage/Patch.cs

```diff
@@ -404,6 +413,16 @@ public void ReplaceSpecies(Species old, Species newSpecies, bool replaceInHistor
         // TODO: can we do something about the game log here?
     }
 
+    public void UpdateAverageSunlight(DayNightCycle lightCycle)
+    {
+        Biome.Sunlight!.Average = Biome.Sunlight.Maximum * lightCycle.AverageSunlight;
```

Comment:
This Sunlight nullability override that's all over really tells me that the design on Biome.Sunlight being nullable is wrong, instead it *should not* be nullable to make other places in the code much easier to reason about.